client/helpers.go

```diff
@@ -55,6 +55,59 @@ func applyChangelist(repo *tuf.Repo, cl changelist.Changelist) error {
 }
 
 func applyTargetsChange(repo *tuf.Repo, c changelist.Change) error {
+	switch c.Type() {
+	case changelist.TypeTargetsTarget:
+		return changeTargetMeta(repo, c)
+	case changelist.TypeTargetsDelegation:
+		return changeTargetsDelegation(repo, c)
+	default:
+		return fmt.Errorf("only target meta and delegations changes supported")
+	}
+}
+
+func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
+	switch c.Action() {
+	case changelist.ActionCreate, changelist.ActionUpdate:
```

Comment:
Yeah, I wasn't sure. This was an effort to do the right thing even in the
face of incoming server changes that may have removed a role from under
you. If 2 admins are both making updates, one deletes a role while the
other is trying to modify it, I figure they're both making valid actions,
although it would be fair to assume the second action may be making some
assumptions about things like the threshold or name (I haven't actually
implemented renaming roles, I'm not planning to support it for friday...),
so possibly it does make sense to separate them.